config-service-impl/src/main/java/org/hypertrace/config/service/store/FilterBuilder.java

```diff
@@ -0,0 +1,105 @@
+package org.hypertrace.config.service.store;
+
+import static org.hypertrace.config.service.store.ConfigDocument.CONFIG_FIELD_NAME;
+
+import io.grpc.Status;
+import org.hypertrace.config.service.v1.LogicalFilter;
+import org.hypertrace.config.service.v1.RelationalFilter;
+import org.hypertrace.core.documentstore.Filter;
+
+public class FilterBuilder {
+
+  public Filter buildDocStoreFilter(org.hypertrace.config.service.v1.Filter filter) {
+    switch (filter.getTypeCase()) {
+      case LOGICAL_FILTER:
+        return evaluateCompositeExpression(filter.getLogicalFilter());
+      case RELATIONAL_FILTER:
+        return evaluateLeafExpression(filter.getRelationalFilter());
+      case TYPE_NOT_SET:
+      default:
+        throw Status.INVALID_ARGUMENT.withDescription("Filter type unset").asRuntimeException();
+    }
+  }
+
+  private Filter evaluateCompositeExpression(LogicalFilter logicalFilter) {
+    switch (logicalFilter.getOperator()) {
+      case LOGICAL_OPERATOR_OR:
+        Filter[] childFilters =
+            logicalFilter.getOperandsList().stream()
+                .map(this::buildDocStoreFilter)
+                .toArray(Filter[]::new);
+        Filter filter = new Filter();
+        filter.setOp(Filter.Op.OR);
+        filter.setChildFilters(childFilters);
+        return filter;
+      case LOGICAL_OPERATOR_AND:
+        Filter[] childFilters2 =
```

Comment:
Could we avoid the `*2` naming on these variables? Either inline or use a local block to avoid collisions.